### PR TITLE
Time endpoint regen after redeploying Cilium in nightly tests.

### PR DESCRIPTION
**Pull request by nebril**
This change adds measurement of regenerating Cilium endpoints after Cilium daemonset is deleted and recreated.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/2166)

### CODE REVIEW

**Comment by nebril** [comment]
We should not merge this until #2167 is resolved

**Comment by ianvernon** [comment]
@nebril since #2167 is resolved, do you want reviews on this? If so, please add the "pending-review" tag and rebase.

**Comment by nebril** [comment]
test-nightly

File: test/k8sT/Nightly.go [hunk 1]
```diff
@@ -112,10 +113,38 @@ var _ = Describe("NightlyEpsMeasurement", func() {
...
+			res := kubectl.Delete(ciliumPath)
+			res.ExpectSuccess(res.GetDebugMessage())
+
+			res = kubectl.Apply(ciliumPath)
+			res.ExpectSuccess(res.GetDebugMessage())
+
+			_, err := kubectl.WaitforPods(helpers.KubeSystemNamespace, "-l k8s-app=cilium", 600)
+			Expect(err).Should(BeNil())
+
+			ciliumPod, err = kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
```
Comment (eloycoto) [inline test/k8sT/Nightly.go]: ATM the nightly test are working with 4 Kubernetes nodes. Could you use the following way please?
```
ciliumPods, err := kubectl.GetCiliumPods(helpers.KubeSystemNamespace)
Expect(err).To(BeNil(), "Cannot retrieve cilium pods")
```

File: test/k8sT/Nightly.go [hunk 2]
```diff
@@ -112,10 +113,38 @@ var _ = Describe("NightlyEpsMeasurement", func() {
...
+			Expect(err).Should(BeNil())
+			ciliumPod2, err = kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s2)
+			Expect(err).Should(BeNil())
+
+			pods = []string{ciliumPod, ciliumPod2}
+			Eventually(func() bool {
+				count := 0
+				for _, pod := range pods {
+					output := kubectl.CiliumExec(pod, "cilium endpoint list").Output().String()
```
Comment (eloycoto) [inline test/k8sT/Nightly.go]: I would like if you can use the following: 
```
count := 0
				for _, pod := range ciliumPods {
					status := kubectl.CiliumEndpointsStatus(pod)
					result := map[string]int{}
					for _, state := range status {
						result[state]++
					}
					count += result[desiredState]
					log.WithField("status", result).Debugf("Cilium %s endpoint status are:", pod)
				}
return count >= endpointCount
```

with this code, we can see the state per each time that runs, so it'll be helpful for debugging in the future.

**Review by eloycoto — CHANGES_REQUESTED** [review]

**Comment by nebril** [comment]
test-nightly

**Comment by joestringer** [comment]
@nebril do we still need this now that https://github.com/cilium/cilium/pull/4684 is merged which exports prometheus metrics for endpoint regeneration time?

**Pull request closed without merging**